src/backend/distributed/transaction/backend_data.c

```diff
@@ -501,3 +501,57 @@ CurrentDistributedTransactionNumber(void)
 
 	return MyBackendData->transactionId.transactionNumber;
 }
+
+
+/*
+ * LockBackendDataSharedMemory locks the shared memory segment used to keep
```

Comment:
Mostly to avoid using spinlocks everywhere in `dump_local_wait_edges`. Will clarify the comment a bit.